### PR TITLE
APP-116: Fix storm demo

### PR DESCRIPTION
**Pull request by Dubland**
/cc @rasendubi @forGGe

### CODE REVIEW

File: stormdemo/source/c/build.sh [hunk 1]
```diff
@@ -31,56 +31,24 @@ fi
...
-function build_app {
-    cd $PROJECT_HOME &&
-    mkdir -p "$PROJECT_HOME/$BUILD_DIR" &&
-    cp "$KAA_LIB_PATH/$BUILD_DIR/"libkaa* "$PROJECT_HOME/$BUILD_DIR/" &&
-    cd $BUILD_DIR &&
-    cmake -DAPP_NAME=$APP_NAME ..
+    mkdir -p "$PROJECT_HOME/$BUILD_DIR"
+    cd $BUILD_DIR
+    cmake -DAPP_NAME=$APP_NAME \
+          -DKAA_MAX_LOG_LEVEL=6 \
```
Comment (ohamov) [inline stormdemo/source/c/build.sh]: please change log level to 3

File: stormdemo/source/c/build.sh [hunk 2]
```diff
@@ -31,56 +31,24 @@ fi
 APP_NAME="demo_client"
 PROJECT_HOME=$(pwd)
 BUILD_DIR="build"
-LIBS_PATH="libs"
 KAA_LIB_PATH="$LIBS_PATH/kaa"
```
Comment (ohamov) [inline stormdemo/source/c/build.sh]: There is no LIBS_PATH

File: stormdemo/source/c/build.sh [hunk 3]
```diff
@@ -31,56 +31,24 @@ fi
...
+    cd $BUILD_DIR
+    cmake -DAPP_NAME=$APP_NAME \
+          -DKAA_MAX_LOG_LEVEL=6 \
+          -DWITH_EXTENSION_NOTIFICATION=OFF ..
     make
 }
 
-function clean {
+clean() {
     rm -rf "$KAA_LIB_PATH/$BUILD_DIR"
```
Comment (ohamov) [inline stormdemo/source/c/build.sh]: redundant command

File: stormdemo/source/c/src/kaa_demo.c [hunk 4]
```diff
@@ -21,12 +21,12 @@
 //#include <string.h>
 //#include <time.h>
```
Comment (ohamov) [inline stormdemo/source/c/src/kaa_demo.c]: remove comment please

File: stormdemo/source/c/build.sh [hunk 5]
```diff
@@ -1,3 +1,4 @@
+#!/bin/bash
```
Comment (ohamov) [inline stormdemo/source/c/build.sh]: sh

File: stormdemo/source/c/build.sh [hunk 6]
```diff
@@ -31,56 +31,24 @@ fi
...
-    cd $PROJECT_HOME &&
-    mkdir -p "$PROJECT_HOME/$BUILD_DIR" &&
-    cp "$KAA_LIB_PATH/$BUILD_DIR/"libkaa* "$PROJECT_HOME/$BUILD_DIR/" &&
-    cd $BUILD_DIR &&
-    cmake -DAPP_NAME=$APP_NAME ..
+    mkdir -p "$PROJECT_HOME/$BUILD_DIR"
+    cd $BUILD_DIR
+    cmake -DAPP_NAME=$APP_NAME \
+          -DKAA_MAX_LOG_LEVEL=6 \
+          -DWITH_EXTENSION_NOTIFICATION=OFF ..
```
Comment (ohamov) [inline stormdemo/source/c/build.sh:43]: disable extension which aren't used

File: stormdemo/source/c/CMakeLists.txt [hunk 7]
```diff
@@ -15,23 +15,22 @@
...
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src
-                    ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/src
-                   )
+execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${KAA_SDK_DIR})
+execute_process(COMMAND sh -c "tar -zxf ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/kaa-c*.tar.gz -C ${KAA_SDK_DIR}")
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -g3 -Wall -Wextra")
 
-link_directories(${CMAKE_CURRENT_SOURCE_DIR}/build)
+set (SAMPLE_SOURCE_FILES src/kaa_demo.c)
```
Comment (ohamov) [inline stormdemo/source/c/CMakeLists.txt]: redundant space

File: stormdemo/source/c/CMakeLists.txt [hunk 8]
```diff
@@ -15,23 +15,22 @@
 #
 
 cmake_minimum_required(VERSION 2.8.8)
-
-project (C-SDK-sample)
-enable_language(C)
+project (C-SDK-sample C)
 
 find_package (OpenSSL REQUIRED)
```
Comment (ohamov) [inline stormdemo/source/c/CMakeLists.txt]: redundant space

File: stormdemo/source/c/CMakeLists.txt [hunk 9]
```diff
@@ -15,23 +15,22 @@
 #
 
 cmake_minimum_required(VERSION 2.8.8)
-
-project (C-SDK-sample)
-enable_language(C)
+project (C-SDK-sample C)
```
Comment (ohamov) [inline stormdemo/source/c/CMakeLists.txt]: remove space

File: stormdemo/source/c/build.sh [hunk 10]
```diff
@@ -31,20 +31,19 @@ fi
...
 
 build_app() {
     cd $PROJECT_HOME
     mkdir -p "$PROJECT_HOME/$BUILD_DIR"
     cd $BUILD_DIR
     cmake -DAPP_NAME=$APP_NAME \
-          -DKAA_MAX_LOG_LEVEL=6 \
+          -DKAA_MAX_LOG_LEVEL=3 \
+          -DWITH_EXTENSION_CONFIGURATION=OFF \
           -DWITH_EXTENSION_NOTIFICATION=OFF ..
```
Comment (ohamov) [inline stormdemo/source/c/build.sh:43]: What about disabling event extension?

File: stormdemo/source/c/CMakeLists.txt [hunk 11]
```diff
@@ -15,23 +15,22 @@
...
+set(KAA_SDK_DIR "client-c")
 
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -g3 -Wall -Wextra")
+execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${KAA_SDK_DIR})
+execute_process(COMMAND sh -c "tar -zxf ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/kaa-c*.tar.gz -C ${KAA_SDK_DIR}")
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src
-                    ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/src
-                   )
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -g3 -Wall -Wextra")
```
Comment (ohamov) [inline stormdemo/source/c/CMakeLists.txt]: use c99 instead of gnu99

**Comment by ohamov** [comment]
LGTM

**Comment by rasendubi** [comment]
LGTM

**Comment by forGGe** [comment]
+

**Pull request merged**